### ISSUE TITLE
Duplicate events when using useExisting providers

### ISSUE THREAD

**Issue by matt1097** [nestjs/event-emitter#578]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When defining providers with useExisting, any OnEvent handlers get registered multiple times with event-emitter

With the repo below, this is the console.log output:

> Sending Event
> Event Received [
>   '4e9f9082-5ad0-4c6d-84d1-8bd036a746e7',
>   '897c020e-7f77-4b6d-b298-579da9025cfe'
> ]
> Event Received [
>   '4e9f9082-5ad0-4c6d-84d1-8bd036a746e7',
>   '897c020e-7f77-4b6d-b298-579da9025cfe'
> ]

This shows that the same event was received by the same object twice.

According to the documentation useExisting is only supposed to create an alias, I do not expect the service to start doing duplicate work.

### Minimum reproduction code

https://github.com/matt1097/nestjs-event-issue

### Steps to reproduce

1. npm i
2. npm run start
3. send the REST call: http://127.0.0.1:3000/test
4. Observe console output

### Expected behavior

The following output is what is expected:

> Sending Event
> Event Received [
>   '4e9f9082-5ad0-4c6d-84d1-8bd036a746e7',
>   '897c020e-7f77-4b6d-b298-579da9025cfe'
> ]

Basically that the event was handled only once

### Package version

1.3.0

### NestJS version

9.0.8

### Node.js version

14.19.3

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

### Other

I can get the expected behavior by modifying the forEach loop here: https://github.com/nestjs/event-emitter/blob/master/lib/event-subscribers.loader.ts#L49

if I add:
```js
if (wrapper.isAlias) {
    return;
}
```
at the beginning of the foreach loop, it will skip registering the event handler a second time and I will get my expected behavior

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by matt1097** [comment]
Created PR

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/event-emitter/pull/579
